oida/commands/componentize.py

```diff
@@ -185,3 +196,140 @@ class {class_name}(AppConfig):
+            ):
+                return tree.visit(self)
+        return tree
+
+    def visit_Decorator(self, node: cst.Decorator) -> Optional[bool]:
+        # Determine if the decorator is the @app.task decorator
+        if m.matches(
+            node,
+            m.Decorator(m.Call(m.Attribute(value=m.Name("app"), attr=m.Name("task")))),
+        ):
+            self.in_app += 1
+        return super().visit_Decorator(node)
+
+    def leave_Decorator(
+        self, original_node: cst.Decorator, updated_node: cst.Decorator
+    ) -> Union[Decorator, FlattenSentinel[Decorator], RemovalSentinel]:
+        # Determine if the decorator is the @app.task decorator
+        if m.matches(
+            original_node,
+            m.Decorator(m.Call(m.Attribute(value=m.Name("app"), attr=m.Name("task")))),
+        ):
+            self.in_app -= 1
+        return updated_node
+
+    def leave_Call(
+        self, original_node: cst.Call, updated_node: cst.Call
+    ) -> cst.BaseExpression:
+        # self.in_app > 0 if we are in an @app.task decorator
+        if self.in_app > 0:
+            # Test if the name of the task is not explicitly set
+            if not m.matches(
+                original_node,
+                m.Call(
+                    args=[m.ZeroOrMore(), m.Arg(keyword=m.Name("name")), m.ZeroOrMore()]
+                ),
+            ):
+                arguments = original_node.args
+                # The implicit name of the task is the path to the old module concatenated with the function name.
+                task_name = f'"{self.old_module}.{self.function_name}"'
+                arguments = (  # type: ignore
+                    cst.Arg(
+                        keyword=cst.Name("name"),
+                        value=cst.SimpleString(value=task_name),
+                        # Equal and comma are not strictly necessary - the code would be functional without these.
+                        # They are needed however to comply with coding conventions and pass the unit test.
+                        equal=cst.AssignEqual(
+                            whitespace_before=cst.SimpleWhitespace(value=""),
+                            whitespace_after=cst.SimpleWhitespace(value=""),
+                        ),
+                        comma=cst.Comma(
+                            whitespace_before=cst.SimpleWhitespace(value=""),
+                            whitespace_after=cst.ParenthesizedWhitespace(
+                                first_line=cst.TrailingWhitespace(
+                                    whitespace=cst.SimpleWhitespace(value=""),
+                                    comment=None,
+                                    newline=cst.Newline(value=None),
+                                ),
+                                empty_lines=[],
+                                indent=True,
+                                last_line=cst.SimpleWhitespace(value="    "),
+                            ),
+                        ),
```

Comment:
Instead of all the whitespace you can just run black on the output, pretty sure there's a helper for that here somewhere?

Reply:
Yes, that happens afterwards. This is only a problem for the unit tests. If I don't add this whitespace, the unit test will fail. If I change the unit test so that the expected outcome also missed the whitespace, the unit test will not go through the Github Actions (black).

Reply:
Ah ok, I'll try that.

Reply:
Yes, it is working!